src/main/java/com/github/maxopoly/MemeMana/model/ManaGainStat.java

```diff
@@ -19,20 +20,30 @@ public ManaGainStat(int streak, long lastDay) {
 	 * @return True if the day has changed since the last check
 	 */
 	public boolean update() {
-		long currentDay = getDayFromTimeStamp(System.currentTimeMillis());
-		if (currentDay == lastDay) {
+		long daysPast = getDayFromTimeStamp(System.currentTimeMillis()) - lastDay;
```

Comment:
See discord

All of this can be done by changing https://github.com/Maxopoly/MemeMana/blob/master/src/main/java/com/github/maxopoly/MemeMana/model/ManaGainStat.java#L29 to

streak = Math.min(0, (int) (streak - currentDay + lastDay));